pgjdbc/src/test/java/org/postgresql/test/jdbc2/RefCursorFetchTest.java

```diff
@@ -6,118 +6,145 @@
-      statement.execute("--create function to read data\n"
-          + "CREATE OR REPLACE FUNCTION test_blob(p_cur OUT REFCURSOR) AS $body$\n"
-          + "BEGIN\n"
-          + "OPEN p_cur FOR SELECT content FROM test_blob;\n"
-          + "END;\n"
-          + "$body$ LANGUAGE plpgsql STABLE");
-
-      statement.execute("--generate 101 rows with 4096 bytes:\n"
-          + "insert into test_blob\n"
-          + "select(select decode(string_agg(lpad(to_hex(width_bucket(random(), 0, 1, 256) - 1), 2, '0'), ''), 'hex')FROM generate_series(1, 4096))\n"
-          + "from generate_series (1, 101)");
+  public RefCursorFetchTest(BinaryMode binaryMode, int numRows, @Nullable Integer fetchSize,
+      AutoCommit autoCommit, boolean commitAfterExecute) {
+    this.numRows = numRows;
+    this.fetchSize = fetchSize;
+    this.autoCommit = autoCommit;
+    this.commitAfterExecute = commitAfterExecute;
+    setBinaryMode(binaryMode);
+  }
 
+  @Parameterized.Parameters(name = "binary = {0}, numRows = {1}, fetchSize = {2}, autoCommit = {3}, commitAfterExecute = {4}")
+  public static Iterable<Object[]> data() {
+    Collection<Object[]> ids = new ArrayList<>();
+    for (BinaryMode binaryMode : BinaryMode.values()) {
+      for(int numRows : new int[]{0, 10, 101}) {
+        for (Integer fetchSize : new Integer[]{null, 0, 9, 50}) {
+          for(AutoCommit autoCommit: AutoCommit.values()) {
+            for(boolean commitAfterExecute: new boolean[]{true, false}) {
+              ids.add(new Object[]{binaryMode, numRows, fetchSize, autoCommit, commitAfterExecute});
+            }
+          }
+        }
+      }
     }
+    return ids;
   }
 
   @Override
-  public void tearDown() throws SQLException {
-    try (Statement stmt = con.createStatement()) {
-      stmt.execute("drop FUNCTION test_blob ()");
-      stmt.execute("drop table test_blob");
+  protected void updateProperties(Properties props) {
+    super.updateProperties(props);
+    if (fetchSize != null) {
+      PGProperty.DEFAULT_ROW_FETCH_SIZE.set(props, fetchSize);
     }
-    super.tearDown();
   }
 
-  @Test
-  public void testRefCursorWithFetchSize() throws SQLException {
-    assumeCallableStatementsSupported();
-    ((PGConnection)con).setDefaultFetchSize(50);
-    int cnt = 0;
-    try (CallableStatement call = con.prepareCall("{? = call test_blob()}")) {
-      con.setAutoCommit(false); // ref cursors only work if auto commit is off
-      call.registerOutParameter(1, Types.REF_CURSOR);
-      call.execute();
-      try (ResultSet rs = (ResultSet) call.getObject(1)) {
-        while (rs.next()) {
-          cnt++;
-        }
-      } catch (SQLException ex) {
-        fail(ex.getMessage());
-      } finally {
-        assertEquals(101, cnt);
+  @BeforeClass
+  public static void beforeClass() throws Exception {
+    try (Connection con = TestUtil.openDB();) {
+      TestUtil.createTable(con, "test_blob", "content bytea");
```

Comment:
`TestUtil.createTable` drops the table if it already exists to avoid the case when "failed test leaving half-populated table"